StortSpelProjekt/Engine/src/Misc/AssetLoader.cpp

```diff
@@ -545,4 +581,94 @@ Font* AssetLoader::loadFont(LPCWSTR filename, int windowWidth, int windowHeight)
 	}
 	Font* font = m_LoadedFonts[filename].first;
 	return m_LoadedFonts[filename].first;
-}
\ No newline at end of file
+}
+void AssetLoader::processAnimations(const aiScene* assimpScene, std::vector<Animation*>* animations)
+{
+	// Store the animations
+	for (unsigned int i = 0; i < assimpScene->mNumAnimations; i++)
+	{
+		Animation* animation = new Animation();
+		aiAnimation* assimpAnimation = assimpScene->mAnimations[i];
+
+		animation->duration = assimpAnimation->mDuration;
+		animation->ticksPerSecond = assimpAnimation->mTicksPerSecond;
+
+		// Store the transform data for each nodeAnimation
+		for (unsigned int j = 0; j < assimpAnimation->mNumChannels; j++)
+		{
+			NodeAnimation nodeAnimation;
+			aiNodeAnim* assimpNodeAnimation = assimpAnimation->mChannels[j];
+			processNodeAnimation(assimpAnimation->mChannels[j], &nodeAnimation);
+			animation->nodeAnimations.push_back(nodeAnimation);
+		}
+
+		// Save the pointer both in the model and the asset loader
+		animations->push_back(animation);
+		m_LoadedAnimations.push_back(animation);
+	}
+}
+
+void AssetLoader::processNodeAnimation(const aiNodeAnim* assimpNodeAnimation, NodeAnimation* nodeAnimation)
+{
+	// possibly do a new here for the nodeanimation
+	// Store the name. This name will be used to know which node (bone) this nodeAnimation belongs to.
+	nodeAnimation->name = assimpNodeAnimation->mNodeName.C_Str();
+
+	// Store the positions
+	for (unsigned int i = 0; i < assimpNodeAnimation->mNumPositionKeys; i++)
+	{
+		nodeAnimation->positions.push_back(
+			DirectX::XMFLOAT3(
+				assimpNodeAnimation->mPositionKeys[i].mValue.x,
+				assimpNodeAnimation->mPositionKeys[i].mValue.y,
+				assimpNodeAnimation->mPositionKeys[i].mValue.z));
+	}
+
+	// Store the rotation quaternions
+	for (unsigned int i = 0; i < assimpNodeAnimation->mNumRotationKeys; i++)
+	{
+		nodeAnimation->rotationQuaternions.push_back(
+			DirectX::XMFLOAT4(
+				assimpNodeAnimation->mRotationKeys[i].mValue.x,
+				assimpNodeAnimation->mRotationKeys[i].mValue.y,
+				assimpNodeAnimation->mRotationKeys[i].mValue.z,
+				assimpNodeAnimation->mRotationKeys[i].mValue.w));
+	}
+
+	// Store the scale values
+	for (unsigned int i = 0; i < assimpNodeAnimation->mNumScalingKeys; i++)
```

Comment:
Do you need three loops of the same size? instead, can you do one loop with three commands?